bikespace_frontend/src/components/submission/location/Location.tsx

```diff
@@ -1,41 +1,68 @@
-import React from 'react';
+import React, {RefObject, useRef, useEffect} from 'react';
+import MapGL, {
+  Marker,
+  AttributionControl,
+  GeolocateControl,
+} from 'react-map-gl/maplibre';
+
 import {LocationLatLng} from '@/interfaces/Submission';
-import {MapContainer, TileLayer, Marker} from 'react-leaflet';
-import {LatLngTuple} from 'leaflet';
 
-import 'leaflet/dist/leaflet.css';
-import 'leaflet-defaulticon-compatibility';
-import 'leaflet-defaulticon-compatibility/dist/leaflet-defaulticon-compatibility.css';
+import 'maplibre-gl/dist/maplibre-gl.css';
 
 import styles from './location.module.scss';
 
 export interface LocationProps {
   location: LocationLatLng;
-  handler: React.ReactNode;
+  setLocation: React.Dispatch<React.SetStateAction<LocationLatLng>>;
 }
 
-function Location({location, handler}: LocationProps) {
-  const position = [location.latitude, location.longitude] as LatLngTuple;
+function Location({location, setLocation}: LocationProps) {
+  const geoControlRef =
+    useRef<maplibregl.GeolocateControl>() as RefObject<maplibregl.GeolocateControl>;
+
+  const {latitude, longitude} = location;
+
+  useEffect(() => {
+    if (!geoControlRef.current) return;
+
+    geoControlRef.current.trigger();
+  }, [geoControlRef.current]);
 
   return (
     <div className={styles.location}>
       <h2>Where was the problem?</h2>
       <h3>Pin the location</h3>
 
       <section className={styles.outerMapContainer} role="application">
-        <MapContainer
-          center={position}
-          zoom={18}
-          scrollWheelZoom={false}
-          style={{height: '100%'}}
+        <MapGL
+          initialViewState={{
+            latitude,
+            longitude,
+            zoom: 16,
+          }}
+          mapStyle="https://basemaps.cartocdn.com/gl/positron-gl-style/style.json"
+          attributionControl={false}
+          onClick={e => {
+            setLocation({
+              latitude: e.lngLat.lat,
+              longitude: e.lngLat.lng,
+            });
+          }}
         >
-          <TileLayer
-            attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
-            url="https://{s}.basemaps.cartocdn.com/light_all/{z}/{x}/{y}{r}.png"
+          <AttributionControl customAttribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors' />
+          <GeolocateControl
+            ref={geoControlRef}
+            position="top-left"
+            trackUserLocation
```

Comment:
When I was testing it, with `trackUserLocation` enabled, the geolocate control didn't seem to work when the user clicks on it. Maybe it needs `onTrackUserLocationStart` and `onTrackUserLocationEnd` to be implemented if this option is used? User click seems to work without `trackUserLocation` enabled, however.